### PR TITLE
Fix DataQuery::exists() not working correctly

### PR DESCRIPTION
**Pull request by 3Dgoo**
Fixes `DataQuery::exists()` not working correctly in some cases as discussed in #9809

### CODE REVIEW

**Review by ScopeyNZ — APPROVED** [review]

**Comment by ScopeyNZ** [comment]
Probably should've got you to rebase that on 4.7...

**Comment by 3Dgoo** [comment]
Do you want us to revert it, rebase it and then merge it in?

**Comment by sminnee** [comment]
Or maybe a core committer can cherry-pick the sha to 4.7

**Comment by ScopeyNZ** [comment]
Yep I'll do a cherry pick. Wasn't at my laptop last night so couldn't. Don't worry about it @3Dgoo - thanks for the help!

**Comment by 3Dgoo** [comment]
Was this fix released with 4.7.1?

**Comment by ScopeyNZ** [comment]
Ah sheesh. I think I did the ol' classic "didn't push"

**Comment by ScopeyNZ** [comment]
I've pushed now though - @sminnee do we still allow making individual patch releases on modules? I can make one or leave it to the team.

**Comment by kinglozzer** [comment]
@ScopeyNZ I was pondering this yesterday too. We have this accepted RFC which says yes: https://github.com/silverstripe/silverstripe-framework/issues/8970. Versioned has broken “lock-step” for patch releases before and that didn’t cause any problems, so I can’t see that anything would’ve changed

**Pull request merged**
